### PR TITLE
Remaining BBox kernel perf optimizations

### PR DESCRIPTION
**Pull request by datumbox**
Some of the opts highlighted at #6872

cc @vfdev-5 @bjuncek @pmeier

### CODE REVIEW

File: torchvision/prototype/transforms/functional/_geometry.py [hunk 1]
```diff
@@ -181,9 +181,11 @@ def resize_bounding_box(
 ) -> Tuple[torch.Tensor, Tuple[int, int]]:
     old_height, old_width = spatial_size
     new_height, new_width = _compute_resized_output_size(spatial_size, size=size, max_size=max_size)
-    ratios = torch.tensor((new_width / old_width, new_height / old_height), device=bounding_box.device)
+    w_ratio = new_width / old_width
+    h_ratio = new_height / old_height
+    ratios = torch.tensor([w_ratio, h_ratio, w_ratio, h_ratio], device=bounding_box.device)
     return (
-        bounding_box.reshape(-1, 2, 2).mul(ratios).to(bounding_box.dtype).reshape(bounding_box.shape),
+        bounding_box.mul(ratios).to(bounding_box.dtype),
```
Comment (datumbox) [inline torchvision/prototype/transforms/functional/_geometry.py:188]: Improvement:
```
[------------ resize cpu torch.float32 ------------]
                |        old       |        new     
1 threads: -----------------------------------------
      (128, 4)  |   13 (+-  0) us  |    8 (+-  0) us
6 threads: -----------------------------------------
      (128, 4)  |   13 (+-  0) us  |    8 (+-  0) us

Times are in microseconds (us).

[----------- resize cuda torch.float32 ------------]
                |        old       |        new     
1 threads: -----------------------------------------
      (128, 4)  |   37 (+-  0) us  |   31 (+-  0) us
6 threads: -----------------------------------------
      (128, 4)  |   37 (+-  0) us  |   31 (+-  0) us

Times are in microseconds (us).

[------------- resize cpu torch.uint8 -------------]
                |        old       |        new     
1 threads: -----------------------------------------
      (128, 4)  |   19 (+-  0) us  |   13 (+-  0) us
6 threads: -----------------------------------------
      (128, 4)  |   19 (+-  0) us  |   13 (+-  0) us

Times are in microseconds (us).

[------------ resize cuda torch.uint8 -------------]
                |        old       |        new     
1 threads: -----------------------------------------
      (128, 4)  |   45 (+-  0) us  |   39 (+-  0) us
6 threads: -----------------------------------------
      (128, 4)  |   45 (+-  0) us  |   39 (+-  1) us

Times are in microseconds (us).
```

File: torchvision/prototype/transforms/functional/_geometry.py [hunk 2]
```diff
@@ -947,7 +951,7 @@ def perspective_bounding_box(
     # 2) Now let's transform the points using perspective matrices
     #   x_out = (coeffs[0] * x + coeffs[1] * y + coeffs[2]) / (coeffs[6] * x + coeffs[7] * y + 1)
     #   y_out = (coeffs[3] * x + coeffs[4] * y + coeffs[5]) / (coeffs[6] * x + coeffs[7] * y + 1)
-
+    # TODO: Investigate potential optimizations by in-placing intermediate results, aminmax etc
```
Comment (pmeier) [inline torchvision/prototype/transforms/functional/_geometry.py:954]: That is already done in #6891.

File: torchvision/prototype/transforms/functional/_geometry.py [hunk 3]
```diff
@@ -1063,7 +1067,7 @@ def elastic_bounding_box(
     # Question (vfdev-5): should we rely on good displacement shape and fetch image size from it
     # Or add spatial_size arg and check displacement shape
     spatial_size = displacement.shape[-3], displacement.shape[-2]
-
+    # TODO: Investigate potential optimizations by in-placing intermediate results, aminmax etc
```
Comment (vfdev-5) [inline torchvision/prototype/transforms/functional/_geometry.py:1070]: This is done in https://github.com/pytorch/vision/pull/6897

**Comment by vfdev-5** [comment]
Maybe, we can merge this after https://github.com/pytorch/vision/pull/6879

**Review by vfdev-5 — APPROVED** [review]
Nice optim for resize, thanks @datumbox

**Comment by datumbox** [comment]
@vfdev-5 I just pushed a couple of untested opts. Could you check again which you think are safe? I'll do benchmarks after we confirm which ones we want in.

File: torchvision/prototype/transforms/functional/_geometry.py [hunk 4]
```diff
@@ -1063,23 +1063,21 @@ def elastic_bounding_box(
...
-    inv_grid = id_grid - displacement
+    inv_grid = id_grid.sub_(displacement)
 
     # Get points from bboxes
-    points = bounding_box[:, [[0, 1], [2, 1], [2, 3], [0, 3]]].reshape(-1, 2)
-    index_x = torch.floor(points[:, 0] + 0.5).to(dtype=torch.long)
-    index_y = torch.floor(points[:, 1] + 0.5).to(dtype=torch.long)
+    points = bounding_box[:, [[0, 1], [2, 1], [2, 3], [0, 3]]].ceil_().reshape(-1, 2)
+    index_x = points[:, 0].to(dtype=torch.long)
+    index_y = points[:, 1].to(dtype=torch.long)
```
Comment (datumbox) [inline torchvision/prototype/transforms/functional/_geometry.py:1074]: My understanding is that you add 0.5 to achieve a ceil effect. I haven't tested this though. Thoughts?

File: torchvision/transforms/functional_tensor.py [hunk 5]
```diff
@@ -940,8 +940,8 @@ def erase(img: Tensor, i: int, j: int, h: int, w: int, v: Tensor, inplace: bool
     return img
 
 
-def _create_identity_grid(size: List[int]) -> Tensor:
-    hw_space = [torch.linspace((-s + 1) / s, (s - 1) / s, s) for s in size]
+def _create_identity_grid(size: List[int], device: torch.device) -> Tensor:
```
Comment (datumbox) [inline torchvision/transforms/functional_tensor.py:943]: I thought to do this directly on V1 to avoid copy-pasting it.

**Comment by vfdev-5** [comment]
I'll cherry pick those for elastic those which makes sense. Thanks for pointers!

File: torchvision/transforms/functional_tensor.py (hunk 5 above)
Comment (vfdev-5) [reply torchvision/transforms/functional_tensor.py:943]: We can do better than this code, for example: https://github.com/pytorch/vision/pull/6897/files#diff-8d715258c77dea35427ef8b4605c5dca131717b7e2f6ca1ee3afa99392813105R1050-R1059

Comment (datumbox) [reply torchvision/transforms/functional_tensor.py:943]: Perfect, reverting this one.

File: torchvision/prototype/transforms/functional/_geometry.py (hunk 4 above)
Comment (vfdev-5) [reply torchvision/prototype/transforms/functional/_geometry.py:1076]: Yes, I'm not sure why I coded like that ...

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @datumbox

**Pull request merged**